libopenage/CMakeLists.txt

```diff
@@ -1,6 +1,20 @@
 # source file configuration
 # for the resulting binary
 
+# filled by pxdgen.
+set_property(GLOBAL PROPERTY SFT_PXDGEN_SOURCES)
+
+function(pxdgen)
+	# add a C++ header file as pxdgen source file
+	foreach(HEADER_SOURCE ${ARGN})
+		if(NOT "${HEADER_SOURCE}" MATCHES ".*\\.h$")
+			message(FATAL_ERROR "pxdgen accepts only .h files. You provided: ${HEADER_SOURCE}")
+		endif()
+
+		get_filename_component(HEADER_SOURCE "${HEADER_SOURCE}" ABSOLUTE)
+		set_property(GLOBAL APPEND PROPERTY SFT_PXDGEN_SOURCES "${HEADER_SOURCE}")
+	endforeach()
+endfunction()
```

Comment:
that function has nothing to do with libopenage, it should be somewhere in the `buildsystem/` folder.